FaunaDB.Client/Client/StreamingEventHandler.cs

```diff
@@ -2,25 +2,61 @@
 using System.Collections.Generic;
 using System.IO;
 using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
 using FaunaDB.Errors;
 using FaunaDB.Types;
 
 namespace FaunaDB.Client
 {
     public class StreamingEventHandler : IObservable<Value>, IDisposable
     {
+        private const int TIME_OUT = 10000;
         private readonly List<IObserver<Value>> observers;
         private StreamReader streamReader;
+        private CancellationTokenSource cancelTokenSource;
+        private Task checkConnectionTask;
 
         private static Field<string> CODE = Field.At("event", "code").To<string>();
         private static Field<string> DESCRIPTION = Field.At("event", "description").To<string>();
         private static Field<string> TYPE = Field.At("type").To<string>();
 
-        public StreamingEventHandler(Stream dataSource)
+        public StreamingEventHandler(Stream dataSource, Func<Task> checkConnection = null)
         {
             dataSource.AssertNotNull(nameof(dataSource));
             observers = new List<IObserver<Value>>();
             streamReader = new StreamReader(dataSource);
+            if (checkConnection != null)
+            {
+                cancelTokenSource = new CancellationTokenSource();
+                CancellationToken token = cancelTokenSource.Token;
+
+                checkConnectionTask = Task.Factory.StartNew(async () =>
+                {
+                    while (true)
+                    {
+                        if (token.IsCancellationRequested)
+                        {
+                            return;
+                        }
+
+                        try
+                        {
+                            await checkConnection?.Invoke();
+                        }
+                        catch (Exception ex)
+                        {
+                            foreach (var observer in observers.ToList())
+                            {
+                                observer.OnError(ex);
+                            }
+                        }
+
+                        Task.Delay(TIME_OUT, token).Wait();
+                    }
+                }, TaskCreationOptions.AttachedToParent
+                );
+            }
```

Comment:
is this formatting right ?

Reply:
Task should be completed before main task. But actually we doesn't need that (It works good without this param)